### ISSUE TITLE
Bug Report - FOMOD: Oddities with the "Required" Option Type

### ISSUE THREAD

**Issue by BellCubeDev** [Nexus-Mods/Vortex#12904]
# **Describe the bug**
* In previous versions of Vortex, "Required" options that were presented as a checkbox in FOMOD Installers were greyed out. However, currently, the checkboxes are not greyed out and are thus presented identically to selected options.
* "Required" options that are presented as radio buttons are not greyed out, either. Furthermore, it would seem that the last option in the list that has a pre-selection type (i.e. "Required" or "Recommended") gets the default selection.

# **To Reproduce**
1. Create or find a FOMOD (like [my testing one](https://gist.github.com/BellCubeDev/bba9a56af3861f5e1d61bdbf3c9888a3)) with a "Required" option that Vortex would display as a checkbox and one that it displays as a radio button
2. Attempt to install the mod/archive
    1. If the checkbox isn't already visible, navigate to its step
3. Attempt to click the checkbox to verify that it is indeed a "Required" option
4. Repeat steps 2.1 and 3 for the radio button
5. See Issue

# **Expected behavior**
* Checkbox: In previous versions of Vortex, the checkbox was greyed out. This was the expected behavior.
* Radio Button: I would expect one of the following:
    1. That the last radio button with the "Required" type would be filled, that all of them would be greyed out, and that any "Recommended" options would be ignored
    2. The same as above, only that all options would be greyed out (my preferred solution)
    3. For Vortex to throw some form of error

# **Screenshots**
![image](https://user-images.githubusercontent.com/33764825/186269983-a65cf603-8dcb-434d-b1d0-84f11689af3c.png)

# **Platform (please complete the following information):**
| Windows Version | Vortex Version |
| :-: | :-: |
| Microsoft Windows 10 Pro, 64-bit<br />21H2 (Build 19044.1889)<br />Windows Feature Experience Pack 120.2212.4180.0 | 1.6.7 |

# **Additional context**
According to DevTools, some styling is still being applied to the checkboxes via the `.checkbox input[type=checkbox]` selector.

**Comment by TanninOne** [comment]
This is not correct, the previous version renders these options in exactly the same way. The code rendering checkboxes hasn't changed in at least 2 years and even before that, required checkboxes weren't rendered as disabled.

I guess the situation around Radio buttons makes more sense if the plugin type is conditional. Having a hard-coded "Required" option in a SelectExactlyOne group makes no sense, if there's no choice there shouldn't be a selection offered. But I can see a use case where it depends on the selection in a different group.

**Comment by BellCubeDev** [comment]
I'm more-or-less raising the SelectExactlyOne as a preemptive measure. I don't have any use case for them myself, and they could certainly be negated with conditions.

As for the checkboxes, it would appear that you are correct. I'm not sure what was happening (could very well be a faulty memory), but I do remember them being rendered as disabled before. I might have to do some more investigation on that one. Either way, some form of visual indicator for read-only options would be ideal.
